hazelcast-client-new/src/test/java/com/hazelcast/client/cache/ClientCacheConfigTest.java

```diff
@@ -170,4 +168,39 @@ public void testGetPreConfiguredCache() {
             Assert.assertNotNull("Cache cannot be retrieved on client: " + i, cache);
         }
     }
+
+    @Test
+    public void createCacheConfigOnAllNodes() {
+        final String CACHE_NAME = "myCache";
+
+        Config config = new Config();
+        CacheSimpleConfig cacheConfig = new CacheSimpleConfig().setName(CACHE_NAME);
+        config.addCacheConfig(cacheConfig);
+
+        // Create servers with configured caches
+        HazelcastInstance server1 = Hazelcast.newHazelcastInstance(config);
+        HazelcastInstance server2 = Hazelcast.newHazelcastInstance(config);
+
+        // Create the hazelcast client instance
+        HazelcastInstance client = HazelcastClient.newHazelcastClient();
+
+        // Create the client cache manager
+        HazelcastClientCachingProvider cachingProvider = HazelcastClientCachingProvider.createCachingProvider(client);
+        CacheManager cacheManager = cachingProvider.getCacheManager();
+
+        Cache<String, String> cache = cacheManager.getCache(CACHE_NAME);
+
+        assertNotNull(cache);
+
+        // First attempt to use the cache will trigger to create its record store
+        for (int i = 0; i < 10; i++) {
```

Comment:
I think we should do better validation that the caches are really created. You can create an operator/runnable and check if the caches exist. 

```
class CheckCacheExistOperator extends AbstractOperation{
	String cachename;

	void run(){
		CacheService service = ...;	
		result = service.exist(cacheName)	
	}
}
```

And then you do something like this:

```
void assertCacheExistsOnAllNoded(String cacheName){
	for(Member member: cluster.getMembers()){
		Future<Boolean> future = operationService.invokeOnMember(null, new CheckCacheExistsOperator(cacheName), getAddress(member));
		assertTrue(future.get());
	}
}
```

And you test would look like this

```
@Test
public void createCacheConfigOnAllNodes() {
	final String CACHE_NAME = "myCache";
	Config config = new Config();
	CacheSimpleConfig cacheConfig = new CacheSimpleConfig().setName(CACHE_NAME);
	config.addCacheConfig(cacheConfig);
	// Create servers with configured caches
	HazelcastInstance server1 = Hazelcast.newHazelcastInstance(config);
	HazelcastInstance server2 = Hazelcast.newHazelcastInstance(config);
	// Create the hazelcast client instance
	HazelcastInstance client = HazelcastClient.newHazelcastClient();
	// Create the client cache manager
	HazelcastClientCachingProvider cachingProvider = HazelcastClientCachingProvider.createCachingProvider(client);
	CacheManager cacheManager = cachingProvider.getCacheManager();
	Cache<String, String> cache = cacheManager.getCache(CACHE_NAME);
	
	assertNotNull(cache);
	assertCacheExistsOnAllNoded(cacheName);
}
```

This way you have really checked that your behaviour works correctly. The current tests doesn't test anything since the cacheManager.getCache(CACHE_NAME) could also fail to create the cache on all members without this test detecting it.